### ISSUE TITLE
[proposal] improve rendering for query without incremental data

### ISSUE THREAD

**Issue by morrys** [relay-tools/relay-hooks#238]
the request comes from [this comment](https://github.com/relay-tools/relay-hooks/issues/146#issuecomment-1419573169):

> Hi @morrys, thanks for explaining the behavior in detail. One follow-up question, is there a way to opt out of partial/incremental rendering? I would prefer having 2 re-renders.

It would be interesting to investigate the possibility of identifying queries that call [next](https://github.com/relay-tools/relay-hooks/blob/master/src/FetchResolver.ts#L174) only once during fetch.

This would allow us to implement automation, such as for [polling](https://github.com/relay-tools/relay-hooks/blob/master/src/FetchResolver.ts#L177), and avoid introducing a new parameter to be passed to useQuery and useLazyLoadQuery.

**Comment by morrys** [comment]
@sibelius do you know what are the use cases when next is called only once? 💯

**Comment by morrys** [comment]
@Lalitha-Iyer I'm considering modifying the onNext function in QueryFetcher like this:

```ts
const onNext = (
    _o: OperationDescriptor,
    snapshot: Snapshot,
    response: GraphQLResponse,
): void => {
    if (!this.snapshot) {
        this.snapshot = snapshot;
        this.subscribe(snapshot);
        this.resolveResult();
        const responses = Array.isArray(response) ? response : [response];
        const isFinal = responses.some((x) => x != null && x.extensions?.is_final === true);
        if (fetchHasReturned && !isFinal) {
            this.forceUpdate();
        }
    }
};
```

So anyone in the relay network can add that information to the response and change the behavior of useQuery.

**Comment by Lalitha-Iyer** [comment]
I would prefer an option over isFinal, just because it's would be easier to build an abstraction on top. Eventually, we could move to the model of automatically figuring out when to skip the extra re-render, and then this option would be used as an escape hatch to force an extra re-render in case unknown bugs come up.

**Comment by morrys** [comment]
released with version 8.0.0
